core/api/src/main/java/com/microsoft/gctoolkit/GCToolKit.java

```diff
@@ -60,17 +51,40 @@ private static JavaVirtualMachine loadJavaVirtualMachine() {
     public GCToolKit() {
         // Allow for adding aggregations from source code,
         // but don't corrupt the ones loaded by the service loader
-        this.registeredAggregations = new HashSet<>(aggregationsFromServiceLoader);
+        this.registeredAggregations = new HashSet<>();
+    }
+
+    /**
+     * Loads all Aggregations defined in the application module through
+     * the java.util.ServiceLoader model. To register a class that
+     * provides the {@link Aggregation} API, define the following
+     * in {@code module-info.java}:
+     * <pre>
+     * import com.microsoft.gctoolkit.aggregator.Aggregation;
+     * import com.microsoft.gctoolkit.sample.aggregation.HeapOccupancyAfterCollectionSummary;
+     * 
+     * module com.microsoft.gctoolkit.sample {
+     *     ...
+     *     provides Aggregation with HeapOccupancyAfterCollectionSummary;
+     * }
+     * </pre>
+     */
+    public void loadAggregationsFromServiceLoader() {
+        if (aggregationsLoadedFromServiceLoader == true) return;
```

Comment:
I don't think this flag/check is necessary. Someone may have reason to call this more than once (dynamically load new aggregations). Unlikely, but possible. Besides, `registeredAggregators` is a set, so you will not end up with more than one entry for a given aggregation class.

Reply:
Interesting thought. Agree it makes sense to simply not have it. Besides, it's not like someone would intentionally call this more than once *unless* they have strong reasons to, as you said. Otherwise it would be a bug in their code.

Reply:
Fixed in 4618237